zkevm-circuits/src/state_circuit/test.rs

```diff
@@ -292,11 +322,79 @@ fn address_limb_out_of_range() {
     assert_error_matches(result, "mpi limb fits into u16");
 }
 
+#[test]
+fn storage_key_mismatch() {
+    let rows = vec![Rw::AccountStorage {
+        rw_counter: 1,
+        is_write: false,
+        account_address: Address::default(),
+        storage_key: U256::from(6),
+        value: U256::zero(),
+        value_prev: U256::zero(),
+        tx_id: 4,
+        committed_value: U256::from(5),
+    }];
+    let overrides = HashMap::from([
+        ((AdviceColumn::StorageKey, 1), Fr::from(10)),
+        (
+            (AdviceColumn::StorageKeyChangeInverse, 1),
+            Fr::from(10).invert().unwrap(),
+        ),
+    ]);
+
+    let result = verify_with_overrides(rows, overrides);
+
+    assert_error_matches(result, "rlc encoded value matches bytes");
+}
+
+#[test]
+fn storage_key_byte_out_of_range() {
+    let rows = vec![Rw::AccountStorage {
+        rw_counter: 1,
+        is_write: false,
+        account_address: Address::default(),
+        storage_key: U256::from(256),
+        value: U256::zero(),
+        value_prev: U256::zero(),
+        tx_id: 4,
+        committed_value: U256::from(5),
+    }];
+    let overrides = HashMap::from([
+        ((AdviceColumn::StorageKey, 1), Fr::from(256)),
```

Comment:
It's needed because the storage key is rlc encoded. The assignment from rows will be `1 * randomness + 0 * 1 = randomness` and not `0 * randomness + 256 * 1 = 256`.